docs-internal/config-metadata.md

```diff
@@ -0,0 +1,404 @@
+     default is `false` (if not defined in document)
+producers (methods in general) - format is "fully qualified type name#method name (method parameter types)", example:
+     `io.helidon.security.providers.common.OutboundTarget.Builder#build()`
+
+### Option element
+Each option is one key in configuration. Option can either be a simple value (String, Long, URI), or a complex value
+defined by another type (nested types).
+
+```json
+{
+     "key": "key in configuration (may be missing, if this merges with parent)",
+     "type": "fully qualified type of the configuration option, defaults to java.lang.String",
+     "description": "description of this configuration node (expected to be non-empty)",
+     "kind": "LIST|MAP|VALUE",
+     "method": "annotated method",
+     "merge": true,
+     "experimental": true,
+     "required": true,
+     "provider": true,
+     "deprecated": true,
+     "defaultValue": "string default value (only for value nodes)"
+     "allowedValues": []
+}
+```
+key - there is a special case when we need to define a list on a method that does not use a nested type
+     in such a case, the key contains `*` to mark a list. Example: `secrets.*.provider`, `secrets.*.config`
+     would result in:
+     ```yaml
+          secrets:
+               - provider: "provider"
+                 config: "something"
+     ```
+type - either a type directly mapped to an object (String, Integer, URI etc.), or a nested type defined either in 
+     this or in another module. Nested configuration may be in another metadata json.
+kind LIST: this is a list of values (either simple values, or objects)
+kind MAP: this is a map of values, using keys of the map as keys, and values of the map as values, `type` defines 
+     the value type (must be a simple value), key is expected to be `java.lang.String`
+kind VALUE: either a simple value (String, Long etc.) or a nested object (this is the default)
+method: method to be called to configure this option (may be a static factory method, such as `create(Config)`, or a builder method)
+merge: if set to `true` (default is `false`) this option's key is ignored and all its nested keys are inserted
+     directly into the parent option
+experimental: experimental options may change without warning even between minor releases (such as support for Loom in Java)
+required: if set to `true` (default is `false`) this option must be present in configuration if parent option is present
+provider: this option expects a type with matching `provides` - such as security providers configuration
+deprecated: this option should no longer be used, there is another option allowing the same (usually when renaming options,
+     fixing inconsistencies etc.); description should provide more details
+defaultValue: for simple values, this defines the string default value
+
+Example of a simple optional element with defa# Configuration metadata
```

Comment:
I think everything up to here is a near-duplicate of what follows. Extra paste?

Reply:
Removed